### PR TITLE
Add support to pyOpenSSL for TLSv1.1 and TLSv1.2

### LINKED ISSUE #695: PyOpenSSL support doesn't work with TLSv1.1 and TLSv1.2
`docker-py` tries to use latest protocol version in the ssl module, so use ssl.PROTOCOL_TLSv1_2. Unfortunately urllib3.contrib.pyopenssl only has SSLv2_3 and TLSv1 so it breaks:

https://github.com/shazow/urllib3/blob/master/urllib3/contrib/pyopenssl.py#L68

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/25970022-pyopenssl-support-doesn-t-work-with-tlsv1-1-and-tlsv1-2?utm_campaign=plugin&utm_content=tracker%2F192525&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F192525&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by sigmavirus24**
Closes #695

### CODE REVIEW

**Comment by sigmavirus24** [comment]
The GAE tests failed trying to import `dev_appserver`.

File: urllib3/contrib/pyopenssl.py [hunk 1]
```diff
@@ -71,6 +71,12 @@
     ssl.PROTOCOL_TLSv1: OpenSSL.SSL.TLSv1_METHOD,
 }
 
+if hasattr(ssl, 'PROTOCOL_TLSv1_1') and hasattr(OpenSSL.SSL, 'TLSv1_1_METHOD'):
+    _openssl_versions[ssl.PROTOCOL_TLSv1_1] = OpenSSL.SSL.TLSv1_1_MEHOD
```
Comment (shazow) [inline urllib3/contrib/pyopenssl.py]: `..._MEHOD`

Would be wise to try this code or something. :P

**Comment by shazow** [comment]
@itamarst If you have a chance, can you confirm that this patch works for you?

File: urllib3/contrib/pyopenssl.py [hunk 2]
```diff
@@ -71,6 +71,12 @@
...
+
+if hasattr(ssl, 'PROTOCOL_TLSv1_2') and hasattr(OpenSSL.SSL, 'TLSv1_2_METHOD'):
+    _openssl_versions[ssl.PROTOCOL_TLSv1_2] = OpenSSL.SSL.TLSv1_2_MEHOD
```
Comment (Lukasa) [inline urllib3/contrib/pyopenssl.py]: Same note as @shazow's here.

File: urllib3/contrib/pyopenssl.py (hunk 1 above)
Comment (sigmavirus24) [inline urllib3/contrib/pyopenssl.py]: lol

File: urllib3/contrib/pyopenssl.py (hunk 2 above)
Comment (sigmavirus24) [inline urllib3/contrib/pyopenssl.py]: Yep, Copy-pasta got me

**Comment by shazow** [comment]
Typos are fixed. Please ping this thread when somebody has had a chance to actually try this code and I'll merge it. :)

**Comment by shazow** [comment]
Btw, would be swell if we had somekind of integration test we could run as a one-off script to verify some features in this contrib. :/

**Comment by Xarthisius** [comment]
@shazow I've tested and it fixes the issues with docker-py. Can we get this merged?

**Comment by shazow** [comment]
Yea looks good. Travis is freaking out right now, investigating in #741. Will merge once that's resolved. :) (Feel free to ping again later)

**Comment by sigmavirus24** [comment]
I kicked this build again to see what would happen.

**Comment by sigmavirus24** [comment]
@shazow passing now

**Pull request merged**